File: backend/microservice/src/level/objects/LevelComment.scala
package microservice.level.objects

import io.circe.generic.semiauto._
import io.circe.{Decoder, Encoder}

final case class LevelComment(
  id: String,
  levelId: String,
  userId: String,
  content: String,
  createdAt: String
)

object LevelComment {
  implicit val encoder: Encoder[LevelComment] = deriveEncoder
  implicit val decoder: Decoder[LevelComment] = deriveDecoder
}

File: frontend/src/objects/level/social/level-comment.ts
import { z } from "zod";

export const LevelCommentSchema = z.object({
  id: z.string().min(1),
  levelId: z.string().min(1),
  userId: z.string().min(1),
  content: z.string().min(1).max(500),
  createdAt: z.string(),
});

export type LevelComment = z.infer<typeof LevelCommentSchema>;
